test/extensions/filters/listener/local_ratelimit/local_ratelimit_test.cc

```diff
@@ -0,0 +1,172 @@
+#include <memory>
+
+#include "source/common/network/io_socket_handle_impl.h"
+#include "source/common/network/listener_filter_buffer_impl.h"
+#include "source/common/stats/isolated_store_impl.h"
+#include "source/extensions/filters/listener/local_ratelimit/local_ratelimit.h"
+
+#include "test/mocks/network/mocks.h"
+#include "test/mocks/stats/mocks.h"
+
+#include "absl/strings/str_format.h"
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+using testing::_;
+using testing::ByMove;
+using testing::InSequence;
+using testing::NiceMock;
+using testing::Return;
+using testing::ReturnRef;
+
+namespace Envoy {
+namespace Extensions {
+namespace ListenerFilters {
+namespace LocalRateLimit {
+namespace {
+
+class LocalRateLimitTest : public testing::Test, public Event::TestUsingSimulatedTime {
+public:
+  class ActiveFilter {
+  public:
+    ActiveFilter(const FilterConfigSharedPtr& config) : filter_(config) {
+      EXPECT_EQ(filter_.maxReadBytes(), 0);
+      ON_CALL(cb_, socket()).WillByDefault(ReturnRef(socket_));
+      ON_CALL(socket_, ioHandle()).WillByDefault(ReturnRef(io_handle_));
+    }
+
+    Network::FilterStatus onAccept() { return filter_.onAccept(cb_); }
+    Network::FilterStatus onEmptyData() {
+      // This is for test coverage purpose only.
+      // onData shouldn't be called since maxReadBytes is zero.
+      Network::ListenerFilterBufferImpl buffer(
+          io_handle_, dispatcher_, [](bool) {}, [](Network::ListenerFilterBuffer&) {}, 1);
+      return filter_.onData(buffer);
+    }
```

Comment:
If you remove those unnecessary overrides from implementation, the function (like this one) and related tests  seems can also be removed?

Reply:
same here.